redhat-appstudio/remote-secret
# fix: propagating the second and next RemoteSecret's updates

### What does this PR do?
1. Setups RemoteSecret reconciler to watch for uploadSecret updates and removals.
2. Removes NotifyingRemoteSecretStorage as it is no longer needed. Switches to secretStorage.
3. Updates integration tests - reconciliation must be triggered manually to pick up the saved data.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?



### How to test this PR?

1. deploy RS operator with img: `quay.io/pbaran/rs:SVPI-539-propagation`
2. create RS and upload a secret (+ target NS): 
```shell
cat <<EOF | kubectl create -n "default" -f -
apiVersion: appstudio.redhat.com/v1beta1
kind: RemoteSecret
metadata:
  name: test-remote-secret
spec:
  secret:
    name: secret-from-remote
  targets:
  - namespace: target-alpha
EOF

cat <<EOF | kubectl create -n "default" -f -
apiVersion: v1
kind: Secret
metadata:
  name: test-remote-secret-secret
  labels:
    appstudio.redhat.com/upload-secret: remotesecret
  annotations:
    appstudio.redhat.com/remotesecret-name: test-remote-secret
type: Opaque
stringData:
  a: b
  c: d
EOF
```
3. Verify that the target secret is deployed and contains correct data.
4. Create new upload secret with different data:
```shell
cat <<EOF | kubectl create -n "default" -f -
apiVersion: v1
kind: Secret
metadata:
  name: test-remote-secret-secret
  labels:
    appstudio.redhat.com/upload-secret: remotesecret
  annotations:
    appstudio.redhat.com/remotesecret-name: test-remote-secret
type: Opaque
stringData:
  foo: bar
EOF
```
5. Verify that target secret contains new data `foo: bar`
